src/main/java/com/rocketden/main/socket/WebSocketConnectionEvents.java

```diff
@@ -81,6 +87,12 @@ public void onSocketConnected(SessionConnectedEvent event) {
             Room room = user.getRoom();
             RoomDto roomDto = RoomMapper.toDto(room);
             socketService.sendSocketUpdate(roomDto);
+
+            // If a game exists, update the room info for that game
+            gameService.conditionallyUpdateSocketInfo(room, user);
```

Comment:
Just want to make sure I understand - why is there a room socket update and a conditional game socket update back-to-back? Is this in the case that someone connects on the "Play Again" page or something, how does the frontend handle these two different types of objects?

Reply:
The previous update (room socket) only updates the room and sends the updated info to the room endpoint (used on the lobby page). So this new method makes sure to also update the game object (if a game has started) to store the updated connection info and send a socket update to the game endpoint (so people on the game page will be able to see if someone connects/disconnects as opposed to just the lobby page).

Reply:
Maybe in the future this could be simplified tho cause yeah the design is definitely not the most intuitive haha